powermock-modules/powermock-module-junit4/src/main/java/org/powermock/modules/junit4/PowerMockRunner.java

```diff
@@ -58,7 +58,16 @@ public void run(RunNotifier notifier) {
         try {
             super.run(notifier);
         } finally {
-            Whitebox.setInternalState(description, "fAnnotations", new Annotation[]{});
+            try {
+               Whitebox.setInternalState(description, "fAnnotations", new Annotation[]{});
```

Comment:
Why the old code made such a drastic operation?
@thekingnothing Is it really needed because it looks like a workaround of something?

Reply:
@Tibor17 it's a "dirty hack" to fix out memory issue. However, I don't see a lot of benefits as annotation does not consume a lot of memory and does not store any state or link to a heavy object.

Reply:
@thekingnothing 
The performance issues are understandable, but I was asking for the purpose why annotations had to be removed from `Description` after Runners execution. Nobody reads them anyway after the runner has finished.

Reply:
Here I cannot help, as this code was introduced 7 years ago and I'm supporting PowerMock only for 3 years. I cannot say the real reason why it was done in such a way.